framework/src/Volo.Abp.AspNetCore/Microsoft/AspNetCore/Builder/AbpApplicationBuilderExtensions.cs

```diff
@@ -18,6 +19,27 @@ public static class AbpApplicationBuilderExtensions
 {
     private const string ExceptionHandlingMiddlewareMarker = "_AbpExceptionHandlingMiddleware_Added";
 
+    public async static Task InitializeApplicationAsync([NotNull] this IApplicationBuilder app)
+    {
+        Check.NotNull(app, nameof(app));
+
+        app.ApplicationServices.GetRequiredService<ObjectAccessor<IApplicationBuilder>>().Value = app;
+        var application = app.ApplicationServices.GetRequiredService<IAbpApplicationWithExternalServiceProvider>();
+        var applicationLifetime = app.ApplicationServices.GetRequiredService<IHostApplicationLifetime>();
+
+        applicationLifetime.ApplicationStopping.Register(() =>
+        {
+            application.ShutdownAsync();
```

Comment:
Calling an async method without await or AsyncHelper.RunSync. Isn't that a problem?

Reply:
code from Microsoft, but I will confirm that.

https://docs.microsoft.com/en-us/dotnet/standard/threading/how-to-register-callbacks-for-cancellation-requests#example

Reply:
https://github.com/abpframework/abp/pull/10928/commits/59284b463ad9483e606618bd53acfaafc7829bc0